### ISSUE TITLE
Version 2.4.1.2 Beta - Link to Author

### ISSUE THREAD

**Issue by oboe45** [openaudible/openaudible#544]
Author Link in the generated Web Page generates a url like "https://www.audible.ca/search?searchAuthor=" - with no text for the author so search does not go to the author page,

**Comment by openaudible** [comment]
Check latest beta 13.
Testing a new method for the author link.

Run full rescan.
Then also Run Refresh Files. 
Then export web page.
Re-open or post new issue if needed.

**Comment by oboe45** [comment]
I did as suggested with new beta 13,
Started by first logging into my audible.ca account.
At start OA Panel says .... 908 books 15 m4b 893 to download - connected
Region CA - 12189 Hours
I Ran full scan  ... which just kept going and going and going
This displayed an Info dialog "Audible Library Update"  – Loaded 908 in
progress 10 queue 898 –
Very slow moving process - what is it doing?  This full scan is really
giving no indication what is happening other than a spinning beach ball! -
started 15:55 Friday  -
Cancelled after 10 minutes - the progress marker moved slower and slower as
time progressed.
then cancelling took another 20 minutes to cancel - after 20 minutes - I
forced the app to quit via the Apple finder, dinner is waiting!!

If it is downloading every one of the books it will take the rest of the
weekend and next week too

This is from the stats.json file after I restarted

{
 "library_last_exception": "java.net.UnknownHostException: api.audible.ca:
nodename nor servname provided, or not known\n\tat
java.base/java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)\n\tat
java.base/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:929)\n\tat
java.base/java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1515)\n\tat
java.base/java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:848)\n\tat
java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1505)\n\tat
java.base/java.net.InetAddress.getAllByName(InetAddress.java:1364)\n\tat
java.base/java.net.InetAddress.getAllByName(InetAddress.java:1298)\n\tat
org.apache.http.impl.conn.SystemDefaultDnsResolver.resolve(SystemDefaultDnsResolver.java:45)\n\tat
org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:112)\n\tat
org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:373)\n\tat
org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:381)\n\tat
org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:237)\n\tat
org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185)\n\tat
org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)\n\tat
org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:111)\n\tat
org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)\n\tat
org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)\n\tat
org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)\n\tat
org.openaudible.audible.api.Client.request(Client.java:103)\n\tat
org.openaudible.audible.api.Client.get(Client.java:181)\n\tat
org.openaudible.audible.api.API.getBooks(API.java:152)\n\tat
org.openaudible.audible.api.API.getBooks(API.java:136)\n\tat
org.openaudible.audible.APIManager.library(APIManager.java:717)\n\tat
org.openaudible.desktop.swt.manager.AudibleGUI$1.run(AudibleGUI.java:357)\n",
 "library_fail": 1,
 "register_success": 19,
 "renew_success": 12,
 "unregister_fail": 1,
 "library_success": 154,
 "unregister_last_exception": "java.lang.NullPointerException\n\tat
org.openaudible.audible.APIManager.unregisterDevice(APIManager.java:192)\n\tat
org.openaudible.desktop.swt.manager.AudibleGUI.logout(AudibleGUI.java:948)\n\tat
org.openaudible.desktop.swt.manager.menu.CommandCenter.execute(CommandCenter.java:343)\n\tat
org.openaudible.desktop.swt.manager.menu.CommandCenter.handleMenuAction(CommandCenter.java:65)\n\tat
org.openaudible.desktop.swt.manager.menu.AppMenu.widgetSelected(AppMenu.java:426)\n\tat
org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:252)\n\tat
org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)\n\tat
org.eclipse.swt.widgets.Display.sendEvent(Display.java:4443)\n\tat
org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1512)\n\tat
org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1535)\n\tat
org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1520)\n\tat
org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1324)\n\tat
org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4229)\n\tat
org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3839)\n\tat
org.openaudible.desktop.swt.gui.GUI.runEventLoop(GUI.java:416)\n\tat
org.openaudible.desktop.swt.gui.GUI.run(GUI.java:516)\n\tat
org.openaudible.desktop.swt.manager.AppLoader.<init>(AppLoader.java:102)\n\tat
org.openaudible.desktop.swt.manager.AppLoader.main(AppLoader.java:112)\n\tat
org.openaudible.desktop.Application.main(Application.java:41)\n\tat
java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native
Method)\n\tat
java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\n\tat
java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n\tat
java.base/java.lang.reflect.Method.invoke(Method.java:567)\n\tat
com.exe4j.runtime.LauncherEngine.launch(LauncherEngine.java:84)\n\tat
com.install4j.runtime.launcher.MacLauncher.main(MacLauncher.java:52)\n",
 "unregister_success": 18
}

Not sure what to do at this point - is this all as expected?

However, suspecting that I had panicked and likely interrupted the process,
I ran "export web page" after reloading OA - at the end of that, all of the
missing Cover pages are now back in place for all of the files, all of the
Author and Narrator links are in place and all work exactly as they
should!!  All that is missing are the m4b links to the converted audio
files - which were likely to be added if I had let this run to completion!
So, its looking really good, therefore I am going to restart the run again
and just let it run to completion.

**Comment by oboe45** [comment]
Sorry if this is killing you with the length of all this, but it seems
close to working.   I restarted the Full Library Sync after deliberately
crashing the previous run.

It went straight into the audible library update at 18:36

Dialog output same as before – a progress counter with meaningful counters
would be reassuring when processing a large library!

After 20 minutes: Dialog stays at “Loaded 908 Books - in progress 10,
Queued 898” – not sure what these numbers reflect in terms of progress

After 34 Minutes: Dialog still stays at “Loaded 908 Books - in progress 10,
Queued 898”

After 43 Minutes: Dialog still stays at “Loaded 908 Books - in progress 10,
Queued 898”

After an hour – the dialog stayed the same as before, SO I killed the app
again. Waited an hour for cancel to finish

Then I killed it again with finder.

I relaunched, exported the web page – it looks the same as it looked two
hours ago – everything is fine except the m4b links are missing –
everything else works OK – so what was the app doing for two hours??

How long should this scan take?  A progress counter which actually shows
movement would be helpful, or perhaps it was just in a massive loop?



Thanks for this - great way to pass the time in a lock down!
